### PR TITLE
Support 3D in ArcLayer

### PR DESCRIPTION
**Pull request by Pessimistress**
For #3869 

arc-lnglat-3d

#### Change List
- Use 3D positions for source/target
- Calculate parabola between 3D points
- Render test
- What's new

### CODE REVIEW

File: test/modules/layers/core-layers.spec.js [hunk 1]
```diff
@@ -102,48 +102,6 @@ test('ArcLayer', t => {
   t.end();
 });
 
-test('ArcLayer#non-iterable', t => {
```
Comment (Pessimistress) [inline test/modules/layers/core-layers.spec.js]: This test targets an edge case in `calculateInstancePositions`, which is removed in this PR.
